### ISSUE TITLE
Incorrect report of chained-comparison

### ISSUE THREAD

**Issue by Lucas-C** [pylint-dev/pylint#10338]
### Bug description

```python
There is some minimal code reproducing the issue:

from types import SimpleNamespace

ns = SimpleNamespace()
ns.x = 0
ns.y = 10

if ns.x >= 2 and ns.y < 2:
    pass
```

### Configuration

```ini

```

### Command used

```shell
pylint pylint_bugrepro.py
```

### Pylint output

```python
************* Module pylint_bugrepro
pylint_bugrepro.py:7:3: R1716: Simplify chained comparison between the operands (chained-comparison)
```

### Expected behavior

Nothing reported by Pylint

### Pylint version

```shell
3.3.6 (latest at this time)
```

### OS / Environment

WSL 2 - Ubuntu 22 LTS

### Additional dependencies

```python

```

**Comment by Pierre-Sassoulas** [comment]
Can't it be simplified as `if ns.x >= 2 > ns.y` ?

**Comment by Lucas-C** [comment]
This is a strange "simplification", but alright, if this is what's expected...
